LTP: fix test case sigrelse01 issue

shaikshavali1 opened the pull request:
This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.
	1. Removed the code related to child process creation.
	2. Moved child and parent-related code/functionality to a single function "verify_sigrelse"
	3. Signal with ID 34 is skipped from testing. This signal ID is used for internal signals in the thread library of sgx-lkl-musl.
	4. Removed pipe handling code which is used for interprocess communication between parent and child.
	5. Removed the code related to the timer, we already have 60sec top-level timer for the test.

shaikshavali1 committed:
LTP: fix test case sigrelse01 issue

This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.
	1. Removed the code related to child process creation.
	2. Moved child and parent-related code/functionality
	   to a single function "verify_sigrelse"
	3. Signal with ID 34 is skipped from testing.
	   This signal ID is used for internal signals in the
	   thread library of sgx-lkl-musl.
	4. Removed pipe handling code which is used for
	   interprocess communication between parent and child.
	5. Removed the code related to the timer,
	   we already have 60sec top-level timer for the test.

davidchisnall commented on tests/ltp/patches/fix_sigrelse_sigrelse01.patch:
```diff
@@ -0,0 +1,673 @@
+ #define DEBUG 0
+ #endif
+-
+-#define CHILD_EXIT(VAL) ((VAL >> 8) & 0377)	/* exit value of child process */
```
I think this is now dead code, please leave it in: eventually we will want to revert these and it makes it much easier to understand the diffs if they only include functional changes.

shaikshavali1 replied:
This patch file will be removed/reverted once #598 is fixed.

davidchisnall commented on tests/ltp/patches/fix_sigrelse_sigrelse01.patch:
```diff
@@ -0,0 +1,673 @@
+ char *TCID = "sigrelse01";	/* test case identifier */
+-static char mesg[MAXMESG];	/* message buffer for tst_res */
+ static int pid;			/* process id of child */
+-static int pipe_fd[2];		/* file descriptors for pipe parent read */
```
We ought to be able to use pipes between threads, why was this change made?

shaikshavali1 replied:
This patch is not using threads. Hence, the pipe related code is removed.

davidchisnall commented on tests/ltp/patches/fix_sigrelse_sigrelse01.patch:
```diff
@@ -0,0 +1,673 @@
+ 	 * parse standard options
+ 	 */
+ 	tst_parse_opts(argc, argv, NULL, NULL);
+-#ifdef UCLINUX
```
Why is this removed?  We should be using the UCLINUX code paths because we are a uCLinux implementation.

shaikshavali1 replied:
uClinux code is removed because exec is the problem and I will update the same in the patch description.

shaikshavali1 replied:
I updated the patch description.

davidchisnall reviewed (changes requested):
This is difficult to review with extraneous changes in the diff, please remove them.

If we are not using the uCLinux changes, please document why and track it (is it because exec is the problem)?

hukoyu commented on tests/ltp/patches/fix_sigrelse_sigrelse01.patch:
```diff
@@ -0,0 +1,673 @@
+This test case creates a child process using "FORK_OR_VFORK".
```
Please change patch file name to sigrelse01.patch 
You can run command
git mv fix_sigrelse_sigrelse01.patch sigrelse01.patch

shaikshavali1 replied:
@hukoyu, I renamed the file to sigrelse01.patch.

hukoyu reviewed (changes requested):
@shaikshavali1  please change the patch file name as requested. Please also address David's feedback.

shaikshavali1 committed:
Rename the file

Rename the file from fix_sigrelse_sigrelse01.patch to sigrelse01.patch

shaikshavali1 commented:
@hukoyu, I addressed your review comment.
@davidchisnall, This patch is not using threads. Hence, the pipe related code is removed. uClinux code is removed because exec is the problem and I will update the same in the commit message.

davidchisnall commented:
Thank you.  Please track this patch as one that can be reverted when #598 is fixed.

shaikshavali1 committed:
Address review comments

Updated patch description in "sigrelse01.patch" file.
added notes on why uClinux code is removed and dependency with
Github issue #598

hukoyu commented:
@davidchisnall  if looks good can you approve so that I can continue to process this change. Thanks

hukoyu commented:
We are making test changes in the https://github.com/lsds/ltp/sgx-lkl now.
This PR resubmitted here: https://github.com/lsds/ltp/pull/2
